packages/admin/admin/src/form/fields/NumberField.tsx

```diff
@@ -3,6 +3,8 @@ import * as React from "react";
 import { Field, FieldProps } from "../Field";
 import { FinalFormNumberInput } from "../FinalFormNumberInput";
 
-export const NumberField = ({ ...restProps }: FieldProps) => {
+export type NumberFieldProps = FieldProps<string, HTMLInputElement>;
```

Comment:
The value has the type number

```suggestion
export type NumberFieldProps = FieldProps<number, HTMLInputElement>;
```